source/common/network/utility.cc

```diff
@@ -110,27 +109,62 @@ uint32_t Utility::portFromUdpUrl(const std::string& url) {
   return portFromUrl(url, UDP_SCHEME, "UDP");
 }
 
-Address::InstanceConstSharedPtr Utility::parseInternetAddress(const std::string& ip_address,
+absl::string_view Utility::hostFromIpAddress(const absl::string_view& authority) {
+  bool ipv6_flag = false;
+  size_t pos = authority.rfind(":");
+
+  if (pos == absl::string_view::npos) {
+    return authority;
+  }
+
+  if (authority[0] == '[') {
+    ipv6_flag = true;
+    pos = authority.rfind("]");
+  } else if (pos != authority.find(":")) {
+    return authority;
```

Comment:
This is malformed host name. Do we actually want to feed it into DNS resolver and see what happens (most likely an error)? Or do we want to make this an error and return 400 to the client?

Reply:
@yanavlasov We should respond with 400 bad request if envoy received not interpretable host name. currently, raise internal envoy exception only.... Is there good solution to resolve this problem?

Reply:
This goes back to the reason for this change. If this change is for #2690 and is not for generic validation of the authority header than 400 may not be the right response. I think this needs to be clarified in the requirements.

Reply:
so I think that this exception should be captured on router and create valid response includes 400. Thank you for your great advice!